misk/src/main/kotlin/misk/hibernate/HibernateHealthCheckModule.kt

```diff
@@ -0,0 +1,30 @@
+package misk.hibernate
+
+import misk.healthchecks.HealthCheck
+import misk.inject.KAbstractModule
+import misk.inject.addMultibinderBinding
+import misk.inject.asSingleton
+import misk.jdbc.DataSourceConfig
+import org.hibernate.SessionFactory
+import java.time.Clock
+import javax.inject.Inject
+import javax.inject.Provider
+
+/**
+ * Binds a HealthCheck for Hibernate.
+ */
+class HibernateHealthCheckModule(
+  private val sessionFactoryProvider: Provider<SessionFactory>,
+  private val config: DataSourceConfig
+) : KAbstractModule() {
+
+  override fun configure() {
+    binder().addMultibinderBinding<HealthCheck>()
+      .toProvider(object : Provider<HibernateHealthCheck> {
```

Comment:
why the manual provider?

Reply:
I wanted to get the SessionFactory lazily, since it's provided by the SessionFactoryService (which needs to be started)

Reply:
can't you just inject a `Provider<SessionFactory>` into the health check?

Reply:
I don't think so, since the `SessionFactory` could be qualified.

Reply:
ah right. thanks!